templates/repo/sub_menu.tmpl

```diff
@@ -39,7 +39,7 @@
 	</div>
 	<a class="ui segment language-stats">
 		{{range .LanguageStats}}
-		<div class="bar" style="width: {{ .Percentage }}%; background-color: {{ .Color }}">&nbsp;</div>
+		<div class="bar" style="width: {{ .Percentage }}%; background-color: {{ .Color }}" title={{ .Language }}>&nbsp;</div>
```

Comment:
```suggestion
		<div class="bar tooltip" style="width: {{ .Percentage }}%; background-color: {{ .Color }}" data-content={{ .Language}}>&nbsp;</div>
```

Reply:
I like that style of that pop-up. But there is one Issue: The pop-up is displayed on the left side which looks not good on large bars. Is there any existing CSS class that displays a centred tooltip or do I need to write my own?

Reply:
Afraid our JS does not support this yet, but it'd be a good addition:

https://github.com/go-gitea/gitea/blob/main/web_src/js/modules/tippy.js#L6

Should be this:

1. Change `placement: 'top-start',` to `placement: target.getAttribute('data-tooltip-placement') || 'top-start',`
2. Add `data-tooltip-placement="top"`

Reply:
This doesn't change anything

Reply:
I'll check later.

Reply:
Works fine for me, please apply this patch:

<img width="1146" alt="image" src="https://user-images.githubusercontent.com/115237/186225498-db321b16-cf50-473d-a99c-6f365541e375.png">


```patch
diff --git a/templates/repo/sub_menu.tmpl b/templates/repo/sub_menu.tmpl
index cbe52386d..157198811 100644
--- a/templates/repo/sub_menu.tmpl
+++ b/templates/repo/sub_menu.tmpl
@@ -38,9 +38,9 @@
 		</div>
 	</div>
 	<a class="ui segment language-stats">
 		{{range .LanguageStats}}
-		<div class="bar" style="width: {{ .Percentage }}%; background-color: {{ .Color }}" title={{ .Language }}>&nbsp;</div>
+		<div class="bar tooltip" style="width: {{ .Percentage }}%; background-color: {{ .Color }}" data-content={{.Language}} data-placement="top">&nbsp;</div>
 		{{end}}
 	</a>
 	{{end}}
 </div>
diff --git a/web_src/js/modules/tippy.js b/web_src/js/modules/tippy.js
index 6f3adaddd..5ab635338 100644
--- a/web_src/js/modules/tippy.js
+++ b/web_src/js/modules/tippy.js
@@ -2,9 +2,9 @@ import tippy from 'tippy.js';

 export function createTippy(target, opts = {}) {
   const instance = tippy(target, {
     appendTo: document.body,
-    placement: 'top-start',
+    placement: target.getAttribute('data-placement') || 'top-start',
     animation: false,
     allowHTML: true,
     maxWidth: 500, // increase over default 350px
     arrow: `<svg width="16" height="7"><path d="m0 7 8-7 8 7Z" class="tippy-svg-arrow-outer"/><path d="m0 8 8-7 8 7Z" class="tippy-svg-arrow-inner"/></svg>`,

Reply:
Now it works for me too. Looks like I accidentality reloaded without cleaning the cache.